### PR TITLE
edit multiple vias

### PR DESCRIPTION
**Pull request by endofexclusive**
Inspired by the tool `Edit hole`.
- Set padstack
- Set parameters
- Set `From rules`
- Can apply to all selected vias for the above

### CODE REVIEW

File: src/dialogs/edit_via.cpp [hunk 1]
```diff
@@ -17,33 +33,102 @@ EditViaDialog::EditViaDialog(Gtk::Window *parent, Via *via, ViaPadstackProvider
...
+    grid->attach(*padstack_apply_all_button, 2, 0, 1, 1);
 
-    button_vp = Gtk::manage(new ViaPadstackButton(vpp));
-    button_vp->property_selected_uuid() = via->vpp_padstack->uuid;
-    button_vp->property_selected_uuid().signal_changed().connect([this, via, &vpp] {
-        if (!via->from_rules)
-            via->vpp_padstack = vpp.get_padstack(button_vp->property_selected_uuid());
+    auto rules_apply_all_button = Gtk::manage(new Gtk::Button);
+    rules_apply_all_button->set_image_from_icon_name("object-select-symbolic", Gtk::ICON_SIZE_BUTTON);
+    rules_apply_all_button->set_tooltip_text("Apply to all pads");
```
Comment (carrotIndustries) [inline src/dialogs/edit_via.cpp]: apply to all vias

File: src/dialogs/edit_via.cpp [hunk 2]
```diff
@@ -17,33 +33,102 @@ EditViaDialog::EditViaDialog(Gtk::Window *parent, Via *via, ViaPadstackProvider
...
+    }
 
-    cb_from_rules = Gtk::manage(new Gtk::CheckButton("From Rules"));
-    cb_from_rules->set_active(via->from_rules);
-    cb_from_rules->signal_toggled().connect([this, via] {
-        via->from_rules = cb_from_rules->get_active();
-        update_sensitive();
+    auto padstack_apply_all_button = Gtk::manage(new Gtk::Button);
+    padstack_apply_all_button->set_image_from_icon_name("object-select-symbolic", Gtk::ICON_SIZE_BUTTON);
+    padstack_apply_all_button->set_tooltip_text("Apply to all pads");
```
Comment (carrotIndustries) [inline src/dialogs/edit_via.cpp]: apply to all vias

File: src/dialogs/edit_via.cpp [hunk 3]
```diff
@@ -17,33 +33,102 @@ EditViaDialog::EditViaDialog(Gtk::Window *parent, Via *via, ViaPadstackProvider
...
     });
-    bbox->pack_start(*button_vp, true, true, 0);
+    grid->attach(*rules_apply_all_button, 2, 1, 1, 1);
+
+    box->pack_start(*grid, false, false, 0);
+
+    std::map<UUID, Via *> viamap;
+
+    for (auto it : vias) {
+        combo->append(static_cast<std::string>(it->uuid), coord_to_string(it->junction->position));
```
Comment (carrotIndustries) [inline src/dialogs/edit_via.cpp]: also add the net name (if there's a net) to the combobox to make it easier to tell tie vias apart

**Comment by carrotIndustries** [comment]
Awesome! I wouldn't have implemented it much differently.

**Comment by endofexclusive** [comment]
Thanks!
The tooltips are now updated and net names added. Also rebased onto the current master.

**Pull request merged**